### PR TITLE
Proxy Radius request with a Kubernetes cluster

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Proxy Radius request with a Kubernetes cluster

### CODE REVIEW

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	"github.com/inverse-inc/go-utils/sharedutils"
+	"github.com/inverse-inc/packetfence/go/chisel/share/cio"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+const radiusAuthK8Filter = "app=radiusd-auth"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: We should allow the environment to override this

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+	namespace := string(data)
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: radiusAuthK8Filter})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	servers := []string{}
+	for _, p := range pods.Items {
+		addr := p.Status.PodIP + ":1812"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: The pod metadata can expose a containerPort to replace the hardcoded value. Or we should at least allow the environment to override the 1812 value

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+	_, controller := cache.NewInformer( // also take a look at NewSharedIndexInformer
+		watchlist,
+		&v1.Pod{},
+		0, //Duration is int64
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				pod := obj.(*v1.Pod)
+				if isPodReady(pod) {
+					address := pod.Status.PodIP + ":1812"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: Same thing about 1812

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+					l.Infof("Adding %s", address)
+					radiusProxy.AddBackend(address)
+					return
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				pod := obj.(*v1.Pod)
+				address := pod.Status.PodIP + ":1812"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: Same thing about 1812

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+				l.Infof("Removing %s", address)
+				radiusProxy.DeleteBackend(address)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				pod := newObj.(*v1.Pod)
+				if isPodReady(pod) {
+					address := pod.Status.PodIP + ":1812"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: Same thing about 1812

File: go/chisel/share/radius_proxy/tunnel_radius.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+					l.Infof("Adding %s", address)
+					radiusProxy.AddBackend(address)
+					return
+				}
+
+				if pod.DeletionTimestamp != nil {
+					address := pod.Status.PodIP + ":1812"
```
Comment (julsemaan) [inline go/chisel/share/radius_proxy/tunnel_radius.go]: Same thing about 1812

File: go/chisel/share/radius_proxy/tunnel_radius.go (hunk 2 above)
Comment (jrouzierinverse) [reply go/chisel/share/radius_proxy/tunnel_radius.go]: Which name should we use to match in the which Port array?

Comment (jrouzierinverse) [reply go/chisel/share/radius_proxy/tunnel_radius.go]: Right now it is called the theport

Comment (julsemaan) [reply go/chisel/share/radius_proxy/tunnel_radius.go]: I'm not sure I follow your thoughts

**Pull request merged**